cpp/include/cudf/strings/char_types/char_types.hpp

```diff
@@ -100,6 +101,50 @@ std::unique_ptr<column> all_characters_of_type(
   string_character_types verify_types = string_character_types::ALL_TYPES,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource());
 
+/**
+ * @brief Filter specific character types from a column of strings.
+ *
+ * To remove all characters of a specific type, set that type in
+ * `types_to_filter` and set `types_to_keep` to `ALL_TYPES`.
+ *
+ * To filter out characters NOT of a select type, specify `ALL_TYPES` for
+ * `types_to_filter` and which types to not remove in `types_to_keep`.
+ *
+ * @code{.pseudo}
+ * Example:
+ * s = ['ab', 'a b', 'a7bb', 'A7B234']
+ * s1 = s.filter_characters_of_type(s,NUMERIC,"",ALL_TYPES)
+ * s1 is ['ab', 'a b', 'abb', 'AB']
+ * s2 = s.filter_characters_of_type(s,ALL_TYPES,"-",LOWER)
+ * s2 is ['ab', 'a-b', 'a-bb', '------']
+ * @endcode
+ *
+ * In `s1` all NUMERIC types have been removed.
+ * In `s2` all non-LOWER types have been replaced.
+ *
+ * One but not both parameters `types_to_filter` and `types_to_keep` must
+ * be set to `ALL_TYPES`.
+ *
+ * Any null row results in a null entry for that row in the output column.
+ *
+ * @throw cudf::logic_error if neither or both `types_to_filter` and
+ *        `types_to_keep` are set to `ALL_TYPES`.
+ *
+ * @param strings Strings instance for this operation.
+ * @param types_to_filter The character types to check in each string.
+ *        Use `ALL_TYPES` here to specify `types_to_keep` instead.
+ * @param types_to_keep Default `ALL_TYPES` means all characters of
+ *        `types_to_filter` will be filtered.
+ * @param mr Device memory resource used to allocate the returned column's device memory.
+ * @return New column of boolean results for each string.
+ */
+std::unique_ptr<column> filter_characters_of_type(
+  strings_column_view const& strings,
+  string_character_types types_to_filter,
```

Comment:
Filter means keep in the Python only because `keep=True` otherwise the filter names mean the same thing in both places. Or maybe I'm misunderstanding your concern.

Reply:
My concern is that "filter" is a generally ambiguous term. :man_shrugging:

Reply:
I see. Maybe `types_to_remove` to go along with `types_to_keep` perhaps.